design-proposals/klip-59-json-functions.md

```diff
@@ -0,0 +1,279 @@
+# KLIP-59 - JSON functions
+
+**Authors**: Aleksandr Sorokoumov (@gerrrr), Colin Hicks (@colinhicks)
+**Release Target**: TBD
+**Status**: _Approved_ |
+**Discussion**: [GitHub PR](https://github.com/confluentinc/ksql/pull/8550)
+
+## Motivation and background
+
+Many ksqlDB users leverage JSON. ETL pipelines often deal in large JSON blobs from, e.g., MongoDB.
+As the de-facto format of the web, it is often the format of analytics events produced to Kafka, as
+well as a choice for materialized views consumed by web apps.
+
+We want to position ksqlDB as a JSON-friendly database. Support for reading, writing, and
+traversing data as JSON presents an important value-add. And users of traditional databases like
+PostgreSQL already enjoy powerful support for JSON.
+
+Today, ksqlDB can:
+* Read JSON-formatted keys and values in input records.
+* Serialize keys and values as JSON in output records.
+* Extract a string value at a given JSON path.
+
+
+## What is in scope
```

Comment:
It'd be great to cover another use case.
You receive a json string (escaped, varchar) and would like to flatten it into the output stream (format=JSON).
I'm not sure it would be possible without introducing a JSON type (union type) or making Map and Array types generic.

Possible alternative would be to have a separate serialiser (say "JSON_FLATTENED").

Previous discussion: https://confluentcommunity.slack.com/archives/C6UJNMY67/p1644308477110799?thread_ts=1644232788.546249&cid=C6UJNMY67

Reply:
@IvanVas could you check and see if there's an issue for this and if not, open one?  I wouldn't want this idea to get lost since this PR is closed.

Reply:
Done, thanks. 
https://github.com/confluentinc/ksql/issues/8747